contrib/mod_tls.c

```diff
@@ -2818,6 +2820,22 @@ static DH *tls_dh_cb(SSL *ssl, int is_export, int keylen) {
           "found matching DH parameter for key length %d", keylen);
         return dhs[i];
       }
+
+      /* Try to find the next "best" DH to use, where "best" means
+       * the smallest DH that is larger than the necessary keylen.
+       */
```

Comment:
This logic should hopefully address the bug, where the principle of least surprise was violated because a DH (4096 bits), larger than the configured server cert (of 2048 bits), was **not** selected.